lib/api/core/models/security/profile.js

```diff
@@ -1,118 +1,143 @@
   highwayhash = require('highwayhash'),
   async = require('async');
 
-function Profile () {
-  this._id = null;
-  this.policies = [];
-}
-
-/**
- * @param {Kuzzle} kuzzle
- *
- * @return {Promise}
- */
-Profile.prototype.getRoles = function profileGetRoles (kuzzle) {
-  var promises;
-  promises = this.policies.map(policy => {
-    return kuzzle.repositories.role.loadRole(policy.roleId)
-      .then(loadedRole => _.assignIn(new Role(), loadedRole, policy));
-  });
-
-  return Promise.all(promises);
-};
-
 /**
- * @param {Request} request
- * @param {Kuzzle} kuzzle
- * @return Promise
+ * @class Profile
  */
-Profile.prototype.isActionAllowed = function profileIsActionAllowed (request, kuzzle) {
-  if (this.policies === undefined || this.policies.length === 0) {
-    return Promise.resolve(false);
+class Profile {
+  constructor() {
+    this._id = null;
+    this.policies = [];
   }
 
-  return this.getRoles(kuzzle)
-    .then(roles => {
-      return new Promise(resolve => {
-        async.some(roles, (role, callback) => {
-          role.isActionAllowed(request, kuzzle)
-            .then(isAllowed => callback(null, isAllowed));
-        }, (error, result) => resolve(result));
-      });
+  /**
+   * @param {Kuzzle} kuzzle
+   *
+   * @return {Promise}
+   */
+  getRoles(kuzzle) {
+    const promises = this.policies.map(policy => {
+      return kuzzle.repositories.role.loadRole(policy.roleId)
+        .then(loadedRole => _.assignIn(new Role(), loadedRole, policy));
     });
-};
 
-/**
- * Validates the Profile format
- *
- * @return {Promise}
- */
-Profile.prototype.validateDefinition = function profileValidateDefinition () {
-  if (!_.isArray(this.policies)) {
-    return Promise.reject(new BadRequestError('The roles member must be an array'));
+    return Bluebird.all(promises);
   }
 
-  if (this.policies.length === 0) {
-    return Promise.reject(new BadRequestError('The roles member array cannot be empty'));
+  /**
+   * @param {Request} request
+   * @param {Kuzzle} kuzzle
+   * @return {Promise<boolean>}
+   */
+  isActionAllowed(request, kuzzle) {
+    if (this.policies === undefined || this.policies.length === 0) {
+      return Bluebird.resolve(false);
+    }
+
+    return this.getRoles(kuzzle)
+      .then(roles => {
+        return new Bluebird(resolve => {
+          async.some(roles, (role, callback) => {
+            role.isActionAllowed(request, kuzzle)
+              .then(isAllowed => callback(null, isAllowed));
+          }, (error, result) => resolve(result));
+        });
+      });
   }
 
-  return Promise.resolve(true);
-};
+  /**
+   * Validates the Profile format
+   *
+   * @return {Promise<boolean>}
+   */
+  validateDefinition() {
+    if (!_.isArray(this.policies)) {
+      return Bluebird.reject(new BadRequestError('The roles member must be an array'));
+    }
+
+    if (this.policies.length === 0) {
+      return Bluebird.reject(new BadRequestError('The roles member array cannot be empty'));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 92 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)